widgets/textinput/textinput.go

```diff
@@ -73,7 +73,7 @@ func New(opts ...Option) (*TextInput, error) {
 		editor: newFieldEditor(),
 		opts:   opt,
 	}
-
+	ti.editor.onChange = opt.onChange
```

Comment:
This introduces a slight layering problem, since the constructor of the TextInput object now understand details of how to construct another object (the field editor).

Can we modify the newFieldEditor() function instead, giving an argument of type OnChangeFn and pass it down on line 73 where newFieldEditor() is called?